lib/miq_expression/target.rb

```diff
@@ -163,6 +163,14 @@ def virtual_attribute?
     false
   end
 
+  def eql?(other)
+    other.kind_of?(MiqExpression::Target) &&
+      column == other.column &&
+      model == other.model &&
+      associations == other.associations
```

Comment:
do associations need to be sorted to ensure they are consistent? (I'm not sure what associations actually is).

Reply:
associations is order dependent. so for `Vm.ems.host-name`, the model is Vm, associations is ['ems', 'host'], and `column` is `name`.

I just want `==` to work (with type conversion), so maybe it is eq and not eql. I thought the type checking seemed close enough

Rspec:

> Ruby exposes several different methods for handling equality:
>
> a.equal?(b) # object identity - a and b refer to the same object
> a.eql?(b) # object equivalence - a and b have the same value
> a == b # object equivalence - a and b have the same value with type conversions]

https://stackoverflow.com/questions/32926817/rspec-eq-vs-eql-in-expect-tests